src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -845,15 +845,75 @@ public FeedbackSessionResultsBundle getFeedbackSessionResultsForStudent(
 
     public String getFeedbackSessionResultsSummaryAsCsv(
             String feedbackSessionName, String courseId,
-            String userEmail, String filterText, boolean isMissingResponsesShown, boolean isStatsShown)
+            String userEmail, String filterText, boolean isMissingResponsesShown,
+            boolean isStatsShown, String questionNumber)
             throws EntityDoesNotExistException, ExceedingRangeException {
 
         return getFeedbackSessionResultsSummaryInSectionAsCsv(
-                feedbackSessionName, courseId, userEmail, null, filterText, isMissingResponsesShown, isStatsShown);
+                feedbackSessionName, courseId, userEmail, null, filterText,
+                isMissingResponsesShown, isStatsShown, questionNumber);
     }
 
     public String getFeedbackSessionResultsSummaryInSectionAsCsv(
             String feedbackSessionName, String courseId, String userEmail,
+            String section, String filterText, boolean isMissingResponsesShown, boolean isStatsShown, String questionNumber)
+            throws EntityDoesNotExistException, ExceedingRangeException {
+
+        long indicatedRange = section == null ? 2000 : -1;
+        FeedbackSessionResultsBundle results = getFeedbackSessionResultsForInstructorInSectionWithinRangeFromView(
```

Comment:
actually here you are retrieving everything regardless of whether it is for all questions or for one questions. This would not improve the performance for the case when you are only downloading one question.  You might need to implement getFeedbackSessionResultsForQuestionForInstructorInSectionWithinRangeFromView and get session for only one question in your database query to really speed up the query or prevent ExceedingRangeException.